### PR TITLE
modify font type and font size for state label text

### PR DESCRIPTION
**Pull request by maria-ericsson**
Signed-off-by: maria-ericsson <maria.chowdhury@ericsson.com>

### CODE REVIEW

**Comment by maria-ericsson** [comment]
For the existing implementation, 
1. State labels are not vertically centered (due to the underscore aligning with the bottom of the state and not showing)
2. On "thin" states the label's text is tiny
Screenshot 2020-09-17 154224

The font is changed to "Verdana" for better readability. Screenshots are here:
for vertically centered
verdana

for tiny text
verdana_tiny_text

File: timeline-chart/src/time-graph-font-controller.ts [hunk 1]
```diff
@@ -5,9 +5,10 @@ export class FontController {
     private fontNameMap: Map<number, Map<string, string>>;
     private fontColorMap: Map<number, string>;
     private defaultFontName: string = "LabelFont8White";
+    private minFonSize: number = 6;
```
Comment (muddana-satish) [inline timeline-chart/src/time-graph-font-controller.ts:8]: We can make this a local variable as we just use it in one function. Also make it constant.

nit: MIN_FONT_SIZE (variable name)

File: timeline-chart/src/time-graph-font-controller.ts [hunk 2]
```diff
@@ -62,6 +63,9 @@ export class FontController {
 
         let fontName: string | undefined;
         if (size) {
+            if(size < this.minFonSize) {
+                size = this.minFonSize;
+            }
```
Comment (muddana-satish) [inline timeline-chart/src/time-graph-font-controller.ts:68]: nit: size = Math.max(size, this.minFonSize)

**Comment by muddana-satish** [comment]
So, the text doesn't overflow if the state's height is too small? (I'm not sure about minimum possible height of the state though).

File: timeline-chart/src/time-graph-font-controller.ts (hunk 2 above)
Comment (maria-ericsson) [reply timeline-chart/src/time-graph-font-controller.ts:68]: Thank you.

File: timeline-chart/src/time-graph-font-controller.ts (hunk 1 above)
Comment (maria-ericsson) [reply timeline-chart/src/time-graph-font-controller.ts:8]: Sure. I will change the name.

**Comment by mad-season** [comment]
@maria-ericsson 

1. In the picture you've attached to show Verdana's centered text, the text in the narrowest states does not appear centered.
2. In the picture you've attached to show readability at reduced sizes, the narrow states shown are not as narrow as can be. Please include a picture with the labels for the narrowest possible state.

File: timeline-chart/src/time-graph-font-controller.ts [hunk 3]
```diff
@@ -5,9 +5,14 @@ export class FontController {
     private fontNameMap: Map<number, Map<string, string>>;
     private fontColorMap: Map<number, string>;
     private defaultFontName: string = "LabelFont8White";
+<<<<<<< 0296bec1d7bad58aea8bdc94e79acd6b479926c3
```
Comment (tahini) [inline timeline-chart/src/time-graph-font-controller.ts:8]: Those are merge conflict markers, please remove

**Comment by tahini** [comment]
For the tiny text, I see it is less tiny than initially, but I'm not sure it's any more readable... but I don't know what can be done about it in such a tiny state. Summoning @ebugden, maybe she has an idea for this.

File: timeline-chart/src/time-graph-font-controller.ts (hunk 3 above)
Comment (maria-ericsson) [reply timeline-chart/src/time-graph-font-controller.ts]: Done. Thank you.

File: timeline-chart/src/time-graph-font-controller.ts [hunk 4]
```diff
@@ -6,8 +6,8 @@ export class FontController {
     private fontColorMap: Map<number, string>;
     private defaultFontName: string = "LabelFont8White";
 
-    constructor(fontFamily: string = "\"Lucida Console\", Monaco, monospace") {
-        this.fontFamily = fontFamily;
+    constructor() {
```
Comment (tahini) [inline timeline-chart/src/time-graph-font-controller.ts:9]: Why not keep it in the constructor as currently, so it's configurable by callers?

**Review by tahini — APPROVED** [review]
No objection, just a question

File: timeline-chart/src/time-graph-font-controller.ts (hunk 4 above)
Comment (maria-ericsson) [reply timeline-chart/src/time-graph-font-controller.ts:9]: That is better. Thank you.

Comment (maria-ericsson) [reply timeline-chart/src/time-graph-font-controller.ts]: Modified the constructor code.

**Review by tahini — APPROVED** [review]
Sounds good

**Review by bhufmann — APPROVED** [review]

**Comment by bhufmann** [comment]
Thanks for the contribution

**Pull request merged**